### PR TITLE
Move contextual data to Problem

### PR DESCRIPTION
**Pull request by donat**

### CODE REVIEW

**Comment by donat** [comment]
@bot-gradle test this

**Comment by reinsch82** [comment]
@bot-gradle test RFN

File: platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/DefaultProblem.java
Comment (hegyibalint) [inline platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/DefaultProblem.java:1]: ❔ Is this class really new? I am suspicious as it has no diff.

Comment (reinsch82) [reply platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/DefaultProblem.java:1]: No, it was changed quite a bit and renamed from `DefaultProblemReport`

Comment (hegyibalint) [reply platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/DefaultProblem.java:1]: Then something wasn't moved? I would expect some red markers, then.

File: platforms/ide/problems-api/src/main/java/org/gradle/api/problems/ProblemId.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ */
+
+package org.gradle.api.problems;
+
+import org.gradle.api.Incubating;
+
+import javax.annotation.Nonnull;
+
+/**
+ * TODO (donat) javadoc.
```
Comment (lkasso) [inline platforms/ide/problems-api/src/main/java/org/gradle/api/problems/ProblemId.java:24]: Javadocs please ;)

File: platforms/ide/problems-api/src/main/java/org/gradle/api/problems/ProblemSpec.java [hunk 2]
```diff
@@ -34,57 +34,41 @@ public interface ProblemSpec {
...
-     *     <li>Should be strictly hierarchical: the category declares the domain and subcategories provide further refinement</li>
-     * </ul>
-     * A few examples with a path-like notation (i.e. {@code category:subcategory1:subcategory2}).
-     * <ul>
-     *     <li>compilation:groovy-dsl</li>
-     *     <li>compilation:java:unused-import</li>
-     *     <li>deprecation:user-code-direct</li>
-     *     <li>task-selection:no-matches</li>
-     * </ul>
+     * TODO.
```
Comment (lkasso) [inline platforms/ide/problems-api/src/main/java/org/gradle/api/problems/ProblemSpec.java:45]: Javadocs please ;)

File: platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/ProblemDefinition.java [hunk 3]
```diff
@@ -34,23 +35,10 @@
...
-     */
-    ProblemCategory getCategory();
-
-    /**
-     * The label of the problem.
-     * <p>
-     * Labels should be short and concise, so they fit approximately in a single line.
-     */
-    String getLabel();
+    ProblemId getId();
```
Comment (lkasso) [inline platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/ProblemDefinition.java:41]: Javadoc TODO?

**Review by lkasso — COMMENTED** [review]
Javadocs requested for review.

**Comment by donat** [comment]
@lkasso This is still under heavy development. I don't know why it was not a draft PR.

**Comment by donat** [comment]
Oh, this is the old one; closing.

**Pull request closed without merging**